components/rollup_node/p2p_interface.md

```diff
@@ -0,0 +1,206 @@
+# P2P Interface
+
+This peer-to-peer (P2P) network service in the rollup node is optional,
```

Comment:
Isn't it also used by all nodes to forward L2 nodes acquired from L1 to the execution engine?

Reply:
I think there should a paragraph explaining what the p2p interface does (e.g. discover peers (& who those are), propagate new L2 sequence blocks, L1 blocks, ...).

I'll also echo my other comment that "P2P Interface" is a sub-optimal name -- an interface is not a component and the organization of the doc is pretty much listing components at the moment. Maybe "P2P layer" or "networking layer"?

Also, this doesn't really describe an interface, but rather the internal organization of sub-components whose interface is specified in other external documents!

Reply:
Actually, an API or programmatic interface to this component would be a really useful thing to have here.

Reply:
> Isn't it also used by all nodes to forward L2 nodes acquired from L1 to the execution engine?

No it's not, P2P is not the same as the engine RPC. Nodes talking to eachother are not regarded as P2P communication in L1 specs, because the relation is fully trusted in the case of RPC, and it's the opposite in P2P.

The "p2p-interface" name is indeed sub-optimal, but already used on L1, so I don't see why we should re-invent the naming here when it's functionally the same.